pkg/distributor/distributor.go

```diff
@@ -1048,7 +1064,7 @@ func (d *Distributor) prepareSeriesKeys(ctx context.Context, req *cortexpb.Write
 		matchedLabelSetLimits := validation.LimitsPerLabelSetsForSeries(limitsPerLabelSet, cortexpb.FromLabelAdaptersToLabels(validatedSeries.Labels))
 		if len(matchedLabelSetLimits) > 0 && labelSetCounters == nil {
 			// TODO: use pool.
-			labelSetCounters = make(map[uint64]*samplesLabelSetEntry, len(matchedLabelSetLimits))
+			labelSetCounters = make(map[uint64]*samplesLabelSetEntry, len(limitsPerLabelSet))
```

Comment:
Is there a reason why now we creating with the `limitsPerLabelSet` size?

Reply:
Umm, I copied over the code I had from last PR and I am a bit lost here.
I think you are right the size should be at max matchedLabelSetLimits. I will update the code.